kvbc/test/v4blockchain/v4_keys_history_test.cpp

```diff
@@ -0,0 +1,319 @@
+// Concord
+//
+// Copyright (c) 2020 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the
+// "License").  You may not use this product except in compliance with the
+// Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the subcomponent's license, as noted in the LICENSE
+// file.
+
+#include "gtest/gtest.h"
+#include "v4blockchain/v4_blockchain.h"
+#include "storage/test/storage_test_common.h"
+#include "bftengine/ReplicaConfig.hpp"
+#include <utility>
+#include <random>
+
+using concord::storage::rocksdb::NativeClient;
+using namespace concord::kvbc;
+using namespace ::testing;
+
+namespace {
+
```

Comment:
Great tests and effort!
Can you please add a test that test shows a flow of a key, for example, if we define key="K":
1 - try to get the version of the K before it was inserted.
2 - insert K at version X
3 - try to get the version of K with version < X
4 - try to get the version of K with version == x 
5 - try to get the version of K with version > x
6 - update K at version Y
7 - try to get the version of K with version < Y and > X
8 - try to get the version of K with version > Y
etc etc

Reply:
Added, thanks!